Source/SwiftLintFramework/Rules/AccessControlOverrideOrderRule.swift

```diff
@@ -0,0 +1,114 @@
+            "↓override open func foo() -> String",
+            "↓override public func foo() -> String",
+            "↓override internal func foo() -> String",
+            "↓override private func foo() -> String"
+        ])
+
+    public func validate(file: File, kind: SwiftDeclarationKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard
+            self.isOverride(attributes: dictionary.enclosedSwiftAttributes),
+            let accessControlLevel = dictionary.accessibility.flatMap(AccessControlLevel.init(identifier:)),
+            let keyOffset = dictionary.offset,
+            let syntaxTokens = self.syntaxTokens(at: keyOffset, in: file) else {
+                return []
+        }
+
+        let attributeTokens = self.attributeTokens(from: syntaxTokens)
+        let contents = self.textContents(of: attributeTokens, in: file)
+
+        guard
+            let accessIndex = contents.index(of: accessControlLevel.description),
+            let overrideIndex = contents.index(of: "override") else {
+                return []
+        }
+
+        if accessIndex > overrideIndex {
+            let offSet = self.safeOffSet(of: attributeTokens[overrideIndex], in: file)
+            return [StyleViolation(ruleDescription: type(of: self).description,
+                                   severity: .error,
+                                   location: Location(file: file, characterOffset: offSet),
+                                   reason: self.configuration.consoleDescription)]
+        }
+
+        return []
+    }
+
+    private func isOverride(attributes: [String]) -> Bool {
+        return attributes.contains("source.decl.attribute.override")
+    }
+
+    private func syntaxTokens(at keywordOffset: Int, in file: File) -> [SyntaxToken]? {
+        return file
+            .syntaxTokensByLine()
+            .flatMap {
+                return $0
+                    .first {
+                        return $0
+                            .contains {
+                                return $0
+                                    .offset == keywordOffset
+                            }
+                    }
+            }
+    }
+
+    private func attributeTokens(from syntaxTokens: [SyntaxToken]) -> [SyntaxToken] {
+        return syntaxTokens.filter { return $0.type == SyntaxKind.attributeBuiltin.rawValue }
+    }
+
+    private func textContents(of attributeTokens: [SyntaxToken], in file: File) -> [String] {
+        return attributeTokens.flatMap {
+            return file.contents.bridge().substringWithByteRange(start: $0.offset, length: $0.length)
+        }
+    }
+
+    private func safeOffSet(of token: SyntaxToken, in file: File) -> Int {
+        let start = token.offset
+        let length = token.length
+        let convertedRange = file.contents.byteRangeToNSRange(start: start, length: length)
+        return convertedRange?.location ?? 0
```

Comment:
isn't the `convertedRange?.location` the same as `token.offset`?

Reply:
I was having trouble getting the proper offset when emojis were in the test files. I had to do this conversion, so it gives the right offset even at that time.